File: Recursion/Other/Easy/Palindrome/Palindrome.cpp
#include<bits/stdc++.h>
using namespace std;

bool isValid(string str, int index) {
    if(index >= str.length()/2) {
        return true;
    }
    if(str[index] != str[str.length() - index - 1]) {
        return false;
    }
    return isValid(str, index + 1);
}
int main(void) {
    cout << isValid("llol", 0);
    return 0;
}

File: Recursion/Other/Easy/Palindrome/Palindrome.java
public class Palindrome {
    static boolean isValid(String s, int index) {
        if(index >= s.length()/2) {
            return true;
        }
        if(s.charAt(index) != s.charAt(s.length() - index - 1)) {
            return false;
        }
        return isValid(s, index + 1);
    }
    public static void main(String[] args) {
        System.out.println(isValid("llol", 0));
    }
}
